### PR TITLE
refactor(page): update `reference-node` using virgo embed

### PR DESCRIPTION
**Pull request by Flrande**

### CODE REVIEW

File: packages/blocks/src/__internal__/rich-text/rich-text.ts [hunk 1]
```diff
@@ -10,8 +10,13 @@ import { isValidUrl } from '../utils/url.js';
...
 import { REFERENCE_NODE } from './reference-node.js';
-import { type AffineTextSchema, type AffineVEditor } from './virgo/types.js';
-const IGNORED_ATTRIBUTES = ['code', 'reference'] as const;
+import {
+  type AffineTextAttributes,
+  type AffineTextSchema,
+  type AffineVEditor,
+} from './virgo/types.js';
+const EDGE_IGNORED_ATTRIBUTES = ['code', 'reference'] as const;
+const GLOBAL_IGNORED_ATTRIBUTES = ['reference'] as const;
```
Comment (doodlewind) [inline packages/blocks/src/__internal__/rich-text/rich-text.ts:19]: Wondering why here requires two levels of ignore?

**Review by doodlewind — APPROVED** [review]

File: packages/blocks/src/__internal__/rich-text/rich-text.ts (hunk 1 above)
Comment (Flrande) [reply packages/blocks/src/__internal__/rich-text/rich-text.ts:19]: It seems that it needs some comments and test, just hold on.

Comment (lawvs) [reply packages/blocks/src/__internal__/rich-text/rich-text.ts:19]: In the beginning, this is a requirement proposed by a product manager.

When you input something after a link or code, the new text should not format as code or link. [PR](https://github.com/toeverything/blocksuite/pull/1822/files#diff-3a54b6d597a9c512f8d8484beffb57b71e7b89bb20085618f270116618f280b5R100-R101)

With the addition of the link automatic detection feature, the linked words are allowed to maintain the hyperlink format (with updated content). [PR](https://github.com/toeverything/blocksuite/pull/2624/files#diff-3a54b6d597a9c512f8d8484beffb57b71e7b89bb20085618f270116618f280b5R14)

After adding the linked page feature, it should not be possible for user input to extend the `reference` format.

File: packages/blocks/src/__internal__/rich-text/reference-node.ts [hunk 2]
```diff
@@ -169,34 +150,17 @@ export class AffineReference extends WithDisposable(ShadowlessElement) {
...
-    // There are some issues if you try to turn off the `contenteditable` attribute in the title node:
-    //   - the cursor may invisible when trying to move across the reference node using the keyboard
-    //
-    // see also [HTML contenteditable with non-editable islands](https://stackoverflow.com/questions/14615551/html-contenteditable-with-non-editable-islands)
-    //
-    // The virgo will skip the zero-width space when calculating the cursor position,
-    // so we use a other zero-width symbol to make the cursor work correctly.
-
-    // This node is under contenteditable="true",
-    // so we should not add any extra white space between HTML tags
```
Comment (lawvs) [inline packages/blocks/src/__internal__/rich-text/reference-node.ts:184]: These comments contain many parts, I think these parts are still useful.

**Review by lawvs — APPROVED** [review]
Thank you very much for helping with the migration. It seems very awesome!

This PR fix https://github.com/toeverything/blocksuite/issues/2006 https://github.com/toeverything/blocksuite/issues/2687

**Pull request merged**